addOns/automation/src/main/java/org/zaproxy/addon/automation/AuthenticationData.java

```diff
@@ -197,6 +207,71 @@ public AuthenticationData(Context context) {
         }
     }
 
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private static void totpMethodToUser(
+            Map<String, Object> parameters, List<ContextWrapper.UserData> users) {
+        if (users == null || users.isEmpty()) {
+            return;
+        }
+
+        var user = users.get(0);
```

Comment:
What about the other users?

Reply:
That will have to wait for changes in core to allow the user to have TOTP data.